core/src/test/java/oracle/weblogic/deploy/util/WLSDeployArchiveTest.java

```diff
@@ -88,6 +91,29 @@ public void testAddRemoveModelWithEmptyZip() throws Exception {
         archive.close();
     }
 
+    @Test
+    public void testAddDirectory() throws Exception {
+        WLSDeployArchive archive = new WLSDeployArchive(APPS_ARCHIVE_FILE_NAME);
+        String appName = archive.addApplication(new File(APP_DIR_TO_ADD));
+        Assert.assertEquals("unexpected app name: " + appName, APP_DIR_ENTRY_NAME, appName);
+        archive.close();
+    }
+
+    @Test
+    public void testIsAFile() throws Exception {
```

Comment:
Please separate directory tests from file tests into new test method "testIsADir", and add assertion for directory found (positive directory test)